### DISCUSSION TITLE
Local settings to override rust-analyzer initialization_options

### DISCUSSION THREAD

**Discussion by darrell-roberts** [zed-industries/zed#8532] [Feature Requests]
Hi,

Is there a way to have folder specific settings for rust-analyzer `initialization_options` in the configuration? I've seen the folder-specific settings [here](https://zed.dev/docs/configuring-zed#folder-specific-settings) however I'd like to also be able to define unique rust-analyzer settings per workspace. I often have different settings for `features` and `excludeDirs` per workspace.

**Comment by Moshyfawn** [comment]
I think the discussion in #4883 would be of interest to you.

**Reply by darrell-roberts** [reply]
Thanks. I left a comment https://github.com/zed-industries/zed/issues/4883#issuecomment-2028872786.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
